Set 'web_readonly_bypass' to installable: True

BT-fgarbely opened the pull request:
I tested the module in my v9 Enterprise Odoo and it worked fine.
This module can be installed in v9 Enterprise.

Commit:
Set 'web_readonly_bypass' to installable: True

pedrobaeza commented:
@BT-fgarbely, thanks for the PR. This module by itself doesn't do anything. Have you tried with a module that uses the feature it has to see if it works?

BT-fgarbely commented:
Hello

I tested it manually on my localhost.
I changed the context of the purchase order menu action according the module description and changed the field 'payment_term_id' to readonly.

This was my test. But I have no module to test this module.

pedrobaeza commented:
OK, that's enough test, thanks.

:+1:

elicoidal commented on README.md:
```diff
@@ -20,6 +20,7 @@ addon | version | summary
 [web_access_rule_buttons](web_access_rule_buttons/) | 9.0.1.0.0 | Disable Edit button if access rules prevent this action
 [web_duplicate_visibility](web_duplicate_visibility/) | 9.0.1.0.0 | Manage the duplicate button visibiliy
 [web_easy_switch_company](web_easy_switch_company/) | 9.0.1.0.0 | Multicompany - Easy Switch Company
+[web_readonly_bypass](web_readonly_bypass/) | 9.0.1.0.0 | Allow to save onchange modifications to readonly fields
```
IMO no need to modify the README.md, this is automatic

pedrobaeza replied:
Yes, please, and make it with `git commit --amend` to avoid a diff in this file that causes conflicts.

tremlin commented:
@BT-fgarbely could you please clean up the commit history? There are a lot of merge commits which is a bit irritating. I think once it's clean, this can be merged.

BT-fgarbely commented:
@tremlin I tried to cleanup the commit history with "git rebase -i <parent>".
But now, I think I created 2 more commits.

Can you tell me what I should do?
Otherwise I can close this PullRequest und create a new "clean" Branch/PullRequest.
What do you think?

tremlin commented:
@BT-fgarbely You could do the following:
- `git rebase -i bbad361^`
- delete all lines in the editor except `pick bbad361`
- edit `__openerp__.py` to bump the version number to `9.0.1.0.0`
- `git add __openerp__.py && git commit --amend`
- check with `git show` and `git log` that you have exactly one commit that realizes the complete changeset
- use `git push -f` to overwrite the branch on the server with your locally updated branch
  If this is too complicated, you could also just create a new branch with your changes and a new PR :)

BT-fgarbely commented:
@tremlin Thanks a lot for your help.
I think, now it's ok  :-)

JonathanNEMRY commented:
WARNING! DO NOT MERGE before I test it!
the JS implementation of the 9.0 has completely changed! I don't think that simply set installable True will make the feature of this module works as it should work...
@BT-fgarbely what did you do as tests?
A correct test can be:
- Debug mode Odoo
- Go into quotation
- Edit form view and make `payement_term` field readonly
- Edit Quotation's action and set context with `{'key': 'value'}`
- Reload the page and click on action.
- Edit the quotation and change customer to modify the payment term
- Add a breakpoint on `instance.web.DataSet.include/write` of `web_readonly_bypass`
- print self.context and it contains as well the `{'key': 'value'}`
  --> the value will be changed by a onchange even if the field is readonly, if this works then OK

JonathanNEMRY commented:
indeed it works fine!
Seems the former method works fine into 9.0 too event if not `odoo.define` defined (https://www.odoo.com/documentation/9.0/reference/javascript.html)
So  :+1:
